ui/ozone/platform/wayland/wayland_data_device.cc

```diff
@@ -39,29 +105,128 @@ void WaylandDataDevice::RequestSelectionData(const std::string& mime_type) {
   // Ensure there is not pending operation to be performed by the compositor,
   // otherwise read(..) can block awaiting data to be sent to pipe.
   read_from_fd_closure_ =
-      base::BindOnce(&WaylandDataDevice::ReadClipboardDataFromFD,
+      base::BindOnce(&WaylandDataDevice::ReadClipboardData,
                      base::Unretained(this), std::move(fd), mime_type);
+  RegisterSyncCallback();
+}
+
+void WaylandDataDevice::RequestDragData(
+    const std::string& mime_type,
+    base::OnceCallback<void(const std::string&)> callback) {
+  base::ScopedFD fd = drag_offer_->Receive(mime_type);
+  if (!fd.is_valid()) {
+    LOG(ERROR) << "Failed to open file descriptor.";
+    return;
+  }
+
+  if (!window_) {
+    LOG(ERROR) << "Could not find window.";
```

Comment:
maybe, there is no sense in calling drag_offer_->Receive if there is no window_.